### ISSUE TITLE
Statefulset prevents upgrade of release

### ISSUE THREAD

**Issue by fischerman** [hashicorp/vault-helm#122]
According to the [docs](https://www.vaultproject.io/docs/platform/k8s/run.html) the recommended way to upgrade Vault is to run a `helm upgrade`.

However, when upgrading from one version to another we get
```
The StatefulSet "vault" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden
```
The template contains `{{ include "vault.chart" . }}` as a label value, which depends on the version of the chart. I believe this is the field that the upgrade tries to patch.

## How to reproduce

```
wget https://github.com/hashicorp/vault-helm/archive/v0.2.1.tar.gz
tar xvzf v0.2.1.tar.gz
helm install vault vault-helm-0.2.1/
nano vault-helm-0.2.1/ # change version field to 0.2.2
helm upgrade --install vault vault-helm-0.2.1/
```
```
helm version
version.BuildInfo{Version:"v3.0.0-rc.3", GitCommit:"2ed206799b451830c68bff30af2a52879b8b937a", GitTreeState:"clean", GoVersion:"go1.13.4"}
```

**Comment by jasonodonnell** [comment]
Hi @fischerman!

That label will need to be removed to prevent this issue from happening again.  As a workaround you can use the `--force` flag which will recreate the statefulset object.  Helm does not delete PVCs (if you aren't using say Consul for storage), so after the upgrade it will just remount the same volume.

**Comment by jasonodonnell** [comment]
Fix proposed here: https://github.com/hashicorp/vault-helm/pull/128

**Comment by jasonodonnell** [comment]
Merged!  Keeping this open until it releases for others to see workaround.

**Comment by fischerman** [comment]
Released with version 0.3.0. Thanks @jasonodonnell
